### PR TITLE
drivers: regulator: read current level from the device

### PR DESCRIPTION
**Pull request by patrickdelaunay**
Always read current voltage level from the device instead of caching the level in struct regulator. This fixes issues for when the regulator level value depends on the parent regulator (supply). It is up the regulator driver to cache or not this value in their private data if applicable.

### CODE REVIEW

File: core/include/drivers/regulator.h [hunk 1]
```diff
@@ -286,10 +284,7 @@ static inline TEE_Result regulator_set_min_voltage(struct regulator *regulator)
  * regulator_get_voltage() - Get regulator current level in microvolt
```
Comment (etienne-lms) [inline core/include/drivers/regulator.h:284]: I think we should rephrase this to `Get regulator effective voltage level in microvolt`.
but that out of the scope of your P-R.

**Review by etienne-lms — COMMENTED** [review]
Makes sense to me.

Nitpicking; reading again the commit message, maybe words "current level" can be confusing; this is about the current _voltage_ level, not the current _current_ level. I would suggest to either replace `current level` with `effective level` or explicit `current voltage level` what do you think? Or maybe:
```
drivers: regulator: do not cache voltage level value

...
```

This this fixes the regulator framework, I think it's worth adding a `Fixes:` tag:
`Fixes: 1a3d3273040b ("drivers: regulator framework")`.

File: core/include/drivers/regulator.h (hunk 1 above)
Comment (patrickdelaunay) [reply core/include/drivers/regulator.h]: done

**Comment by patrickdelaunay** [comment]
> drivers: regulator: do not cache voltage level value

done 

> 
> This this fixes the regulator framework, I think it's worth adding a `Fixes:` tag: `Fixes: 1a3d3273040b ("drivers: regulator framework")`.

done

**Review by etienne-lms — COMMENTED** [review]
`Reviewed-by: Etienne Carriere <etienne.carriere@foss.st.com>` with minor fix in commit message:
*"It is up the regulator driver to cache or not this value in **its** private data..."*
or
*"It is up the regulator driver**s** to cache or not this value in their private data..."*

**Comment by patrickdelaunay** [comment]
>"It is up the regulator driver**s** to cache or not this value in their private data..."_

done
thanks for the review.

**Comment by jforissier** [comment]
> @jforissier, is this change ok to be merged?

It is, sorry for not noticing.
